discovery/syncer.go

```diff
@@ -750,17 +753,18 @@ func (g *GossipSyncer) genChanRangeQuery(
 		startHeight = uint32(newestChan.BlockHeight - chanRangeQueryBuffer)
 	}
 
+	bestHeight := g.cfg.bestHeight()
 	log.Infof("GossipSyncer(%x): requesting new chans from height=%v "+
 		"and %v blocks after", g.cfg.peerPub[:], startHeight,
-		math.MaxUint32-startHeight)
+		bestHeight-startHeight)
 
 	// Finally, we'll craft the channel range query, using our starting
 	// height, then asking for all known channels to the foreseeable end of
 	// the main chain.
 	return &lnwire.QueryChannelRange{
 		ChainHash:        g.cfg.chainHash,
 		FirstBlockHeight: startHeight,
-		NumBlocks:        math.MaxUint32 - startHeight,
+		NumBlocks:        bestHeight - startHeight,
```

Comment:
So from reading the commit message, this change actually isn't required at all? IIRC CL also sends a max uint32 num blocks in their requests. Alternatively, could've we just use the fact that the num blocks message extends beyond our current best known chain tip as a termination condition?

Reply:
Indeed, it's not needed after all. Termination is now based on whether the last reply received satisfies the queried range end height.